File: 0041-first-missing-positive/0041-first-missing-positive.java
class Solution {
    public int firstMissingPositive(int[] nums) {
        int size = nums.length;
        for(int i = 0;i<size;i++){
            while(nums[i]>0 && nums[i]<=size && nums[nums[i]-1]!=nums[i]){
                int tmp = nums[nums[i] - 1];
                nums[nums[i] - 1] = nums[i];
                nums[i] = tmp;


            }
        }
        for(int i = 0;i<size;i++){
            if(nums[i]!=i+1){
                return i+1;
            }
        }
        return size+1;
        
    }
}

File: 0041-first-missing-positive/0041-first-missing-positive.py
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        n = len(nums)
        for i in range(n):
            while n > nums[i] and nums[i] > 0 and nums[i] != nums[nums[i]-1]:
                t = nums[nums[i]-1]
                nums[nums[i]-1] =nums[i] 
                nums[i] = t
        for i in range(n):
            if nums[i] != i+1:
                return i+1
        return n+1
        